src/Compilers/Core/Portable/DiagnosticAnalyzer/DiagnosticAnalysisContext.cs

```diff
@@ -196,6 +197,27 @@ public virtual void ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags an
         {
             throw new NotImplementedException();
         }
+
+        /// <summary>
+        /// Attempts to compute or get the cached value provided by the given <paramref name="valueProvider"/> for the given <paramref name="key"/>.
+        /// Reusing the same <paramref name="valueProvider"/> instance across analyzer actions and/or analyzer instances can improve the overall analyzer performance by avoiding recomputation of the values.
+        /// </summary>
+        /// <typeparam name="TValue">The type of the value associated with the key.</typeparam>
+        /// <param name="key"><see cref="SourceText"/> for which the value is queried.</param>
+        /// <param name="valueProvider">Provider that computes the underlying value associated with the key.</param>
+        /// <param name="value">Value associated with the key.</param>
+        /// <returns>Returns true on success, false otherwise.</returns>
+        public bool TryGetValue<TValue>(SourceText key, SourceTextValueProvider<TValue> valueProvider, out TValue value)
```

Comment:
The SourceText here is not, strictly speaking, a key. It's part of a key (with the value provider being the other part), and it's also input to the computation. Maybe name the parameter something like "text"? The doc comments are also potentially confusing, because they tread the SourceText as if it were a key in a query.